include/engine/engine_config.hpp

```diff
@@ -41,24 +42,9 @@ namespace engine
 
 struct EngineConfig
 {
-    EngineConfig() = default;
-
-    EngineConfig(const boost::filesystem::path &base)
-        : server_paths{{"ramindex", base.string() + ".ramIndex"},
-                       {"fileindex", base.string() + ".fileIndex"},
-                       {"hsgrdata", base.string() + ".hsgr"},
-                       {"nodesdata", base.string() + ".nodes"},
-                       {"edgesdata", base.string() + ".edges"},
-                       {"coredata", base.string() + ".core"},
-                       {"geometries", base.string() + ".geometry"},
-                       {"timestamp", base.string() + ".timestamp"},
-                       {"datasource_names", base.string() + ".datasource_names"},
-                       {"datasource_indexes", base.string() + ".datasource_indexes"},
-                       {"namesdata", base.string() + ".names"}}
-    {
-    }
+    bool IsValid() const;
```

Comment:
We need this for the same reason, we need the `IsValid()` for `*Parameters`. The object is filled element-by element (by boost program options). Also you can modify the member variables since they are public - giving no guarantees at all.